packages/ra-core/src/form/useUnique.spec.tsx

```diff
@@ -0,0 +1,187 @@
+        });
+        render(<WithMessage dataProvider={dataProvider} />);
+
+        await screen.findByDisplayValue('John Doe');
+        fireEvent.change(screen.getByDisplayValue('John Doe'), {
+            target: { value: 'Jordan Doe' },
+        });
+
+        fireEvent.click(screen.getByText('Submit'));
+        await waitFor(() => {
+            expect(dataProvider.create).toHaveBeenCalled();
+        });
+        expect(
+            screen.queryByText('Someone is already registered with this name')
+        ).toBeNull();
+    });
+
+    it('should call the dataProvider with additional filter when provided', async () => {
+        const dataProvider = baseDataProvider();
+        render(<WithAdditionalFilters dataProvider={dataProvider} />);
+
+        await screen.findByDisplayValue('John Doe');
+
+        fireEvent.click(screen.getByText('Submit'));
+
+        await waitFor(() => {
+            expect(dataProvider.getList).toHaveBeenCalledWith('users', {
+                filter: {
+                    name: 'John Doe',
+                    organization_id: 1,
+                },
+                pagination: {
+                    page: 1,
+                    perPage: 1,
+                },
+                sort: {
+                    field: 'id',
+                    order: 'ASC',
+                },
+            });
+        });
+        await screen.findByText('Must be unique');
+    });
+
+    it('should work with deep paths', async () => {
+        const dataProvider = baseDataProvider();
+        render(<DeepField dataProvider={dataProvider} />);
+
+        await screen.findByDisplayValue('John Doe');
+
+        fireEvent.click(screen.getByText('Submit'));
+
+        await waitFor(() => {
+            expect(dataProvider.getList).toHaveBeenCalledWith('users', {
+                filter: {
+                    identity: {
+                        name: 'John Doe',
+                    },
+                },
+                pagination: {
+                    page: 1,
+                    perPage: 1,
+                },
+                sort: {
+                    field: 'id',
+                    order: 'ASC',
+                },
+            });
+        });
+        await screen.findByText('Must be unique');
+        expect(dataProvider.create).not.toHaveBeenCalled();
+    });
+
+    it('should show an error when the dataProvider fails', async () => {
+        render(<DataProviderErrorOnValidation />);
+
+        await screen.findByDisplayValue('John Doe');
+        // The dataProvider for this story fails one over two times
+        // Here's the first time, it should show an error
+        fireEvent.click(screen.getByText('Submit'));
+        expect(screen.queryByText('Server communication error')).toBeNull();
+
+        // Here's the second time, it should show the validation message
+        fireEvent.click(screen.getByText('Submit'));
+        expect(screen.queryByText('Must be unique')).toBeNull();
+    });
+});
```

Comment:
We only have unit tests - I miss an integration test. Could you maybe update the simple example to use the new validator?

Reply:
Added